### PR TITLE
Replaced DetailsFragment by Bottomsheet

### LINKED ISSUE #5: Replace DetailsFragment by Bottom sheet
**Describe the feature you'd like**
Replace the details fragment by a bottom sheet to show the details of a profile.

**Would you like to work on the issue?**

- [ ] Yes
- [ ] No
- Other: Open for all

### PR DESCRIPTION
**Pull request by coder2699**
Fixes #5

Changes: Replaced DetailsFragment by Bottomsheet

Screenshots for the change:

<table>
  <tr>
    <td></td>
    <td></td>
    <td></td>
</tr>
</table>

### CODE REVIEW

File: app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt [hunk 1]
```diff
@@ -68,11 +66,9 @@ class ProfileListAdapter(
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
-                val navOptions = NavOptions.Builder().setEnterAnim(R.anim.nav_default_enter_anim).setExitAnim(
-                    R.anim.nav_default_exit_anim
-                ).setPopEnterAnim(R.anim.nav_default_pop_enter_anim).setPopExitAnim(R.anim.nav_default_pop_exit_anim)
-                    .build()
-                Navigation.findNavController(it).navigate(R.id.detailsFragment, args, navOptions)
+                val dialog = DetailsFragment()
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt]: @coder2699 Please try to create the fragment instance using the sample code image shared on the Gitter channel as it is the recommended way of creating a fragment instance when bundles are also involved.

**Comment by coder2699** [comment]
Still working on it, this is not the finished PR

**Comment by coder2699** [comment]
@rob729 if I am not committing gradle.properties change that I had mentioned earlier, Travis CI won't succeed (if I do, it is succeeding). Though the app is working as desired. What should I do now?

**Comment by rob729** [comment]
Check the error log of Travis CI. You will find the reason why it is failing then fix it accordingly.

File: app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt [hunk 2]
```diff
@@ -66,13 +67,14 @@ class ProfileListAdapter(
...
 
             binding.profileCard.setOnClickListener {
-                val args = Bundle()
-                args.putParcelable("Profile", item)
-                val navOptions = NavOptions.Builder().setEnterAnim(R.anim.nav_default_enter_anim).setExitAnim(
-                    R.anim.nav_default_exit_anim
-                ).setPopEnterAnim(R.anim.nav_default_pop_enter_anim).setPopExitAnim(R.anim.nav_default_pop_exit_anim)
-                    .build()
-                Navigation.findNavController(it).navigate(R.id.detailsFragment, args, navOptions)
+                fun newInstance() = DetailsFragment().apply {
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt]: You don't need to define a function inside the onClickListener.
Have a look at this article(https://blog.mindorks.com/implementing-dialog-fragment-in-android) and see how it is done.
It is not exactly based on this topic so skip the initial part and see how they define the instance of a fragment.

File: app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt [hunk 3]
```diff
@@ -29,7 +27,15 @@ class ProfileListAdapter(
     ListAdapter<Profile, ProfileListAdapter.ViewHolder>(
         ProfileDiffCallbacks()
     ) {
-
+    companion object {
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt]: shift the companion object to Details Fragment as it does not make sense to put this code here.
Also, change the TAG value to something relevant.

File: app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt [hunk 4]
```diff
@@ -29,7 +27,15 @@ class ProfileListAdapter(
...
+        const val TAG = "SimpleDialog"
+        fun newInstance() = DetailsFragment().apply {
+            val args = Bundle()
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt]: Take the args variable as the function parameter.

**Review by rob729 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt (hunk 4 above)
Comment (coder2699) [reply app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt]: @rob729  Took the args variable as function parameter, as you suggested

File: app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt (hunk 3 above)
Comment (coder2699) [reply app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt]: @rob729 I have shifted companion object to Details Fragment.
I have removed "TAG" as it was an extra variable that was not being used anywhere.

**Comment by coder2699** [comment]
I have fixed the extra change that was made in gradle.properties. It won't be a problem now.

File: app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt [hunk 5]
```diff
@@ -68,11 +66,8 @@ class ProfileListAdapter(
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
-                val navOptions = NavOptions.Builder().setEnterAnim(R.anim.nav_default_enter_anim).setExitAnim(
-                    R.anim.nav_default_exit_anim
-                ).setPopEnterAnim(R.anim.nav_default_pop_enter_anim).setPopExitAnim(R.anim.nav_default_pop_exit_anim)
-                    .build()
-                Navigation.findNavController(it).navigate(R.id.detailsFragment, args, navOptions)
+                val dialog = newInstance(args)
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt]: One small change.
change newInstance() to DetailsFragment.newInstance().

Comment (coder2699) [reply app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt]: @rob729  Done 👍

**Review by rob729 — APPROVED** [review]

**Pull request merged**